static_frame/core/util.py

```diff
@@ -3186,5 +3187,38 @@ def iloc_to_insertion_iloc(key: int, size: int) -> int:
         raise IndexError(f'index {key} out of range for length {size} container.')
     return key % size
 
+def _unsized_children(obj):
+    # Check if iterable or a string first for fewer isinstance calls on common types
+    if hasattr(obj, '__iter__') and not isinstance(obj, str):
+        if (
+            (obj.__class__ is np.ndarray and obj.dtype.kind == DTYPE_OBJECT_KIND)
+            or isinstance(obj, tuple)
+            or isinstance(obj, list)
+            or isinstance(obj, set)
+            or isinstance(obj, frozenset)
+        ):
+            yield from obj
+        elif isinstance(obj, dict):
+            yield from chain.from_iterable(obj.items())
+        else:
+            # The full size of the object is included in a getsizeof call
+            # e.g. FrozenAutoMap, integer numpy arrays, int, float, etc.
+            pass
 
-
+def _nested_sizable_elements(obj: any, *, seen=None) -> tp.Iterable[any]:
+    seen = set() if seen is None else seen
+    if id(obj) in seen:
+        return
+    seen.add(id(obj))
+    for el in _unsized_children(obj):
+        yield from _nested_sizable_elements(el, seen=seen)
+    yield obj
+
+def getsizeof_recursive(obj: any, *, seen=None) -> int:
+    seen = set() if seen is None else seen
+    return sum(getsizeof(el) for el in _nested_sizable_elements(obj, seen=seen))
```

Comment:
This is the kind of collection of functions you can put in a class-only Class that groups these functions (as static and class methods) together under a common namespace.

Reply:
Moved the helper functions to `MemoryMeasurements` but kept `getsizeof_total` outside of the class